crate.go

```diff
@@ -92,24 +98,25 @@ func newCrateEndpoint(ep *endpointConfig) *crateEndpoint {
 		return err
 	}
 	return &crateEndpoint{
-		poolConf:     poolConf,
-		readTimeout:  time.Duration(ep.ReadTimeout) * time.Second,
-		writeTimeout: time.Duration(ep.WriteTimeout) * time.Second,
+		poolConf:      poolConf,
+		readPoolSize:  ep.ReadPoolSize,
+		writePoolSize: ep.WritePoolSize,
+		readTimeout:   time.Duration(ep.ReadTimeout) * time.Second,
+		writeTimeout:  time.Duration(ep.WriteTimeout) * time.Second,
 	}
 }
 
 func (c *crateEndpoint) endpoint() endpoint.Endpoint {
+	/**
+	 * Initialize connection pools lazily here instead of in `newCrateEndpoint()`,
+	 * so that the adapter does not crash on startup if the endpoint is unavailable.
+	**/
 	return func(ctx context.Context, request interface{}) (response interface{}, err error) {
```

Comment:
Added the comment about lazy initialization back here, to the spot where it was coming from. The delayed intialization seems to be implemented by returning a function closure here, that detail did not change.

The pool creation has just been refactored into the `createPools()` method, and this one is still called from within the function closure.